### ISSUE TITLE
Delete by date doesn't work

### ISSUE THREAD

**Issue by LoonyBeLuna** [victornpb/undiscord#544]
### Undiscord Version

v5.2.0

### Browser

116.0.3 Firefox

### Extension

TamperMonkey

### Reproduction steps

Set before date to a date in the past.

### Actual results

It deletes all, beginning from newest.

### Expected results

I want to delete ONLY old messages, not the newest.

### Bug affects

- [X] DMs
- [ ] DMs (group)
- [X] Channels
- [ ] Channel (NSFW)

### Additional information & file uploads

![Screenshot_175411-911](https://github.com/victornpb/undiscord/assets/131197997/c9ae4f76-4922-4917-b274-07edad0b0fcb)
![Screenshot_175830-082](https://github.com/victornpb/undiscord/assets/131197997/5fd30d59-965a-4f04-86b6-8ec21a597077)

I don't know how to use the Date correct, I would prefer that I can also decide to delete FIRST the old messages and not the new messages.
But it does ignore the date, always in all my tests.

**Comment by ipaqmaster** [comment]
Yep. Delete by date just deleted all the recent messages and then stopped despite the queue having over 4000 messages. The script's output looks like there was no problem.

Even after wiping the date the script kept saying it had nothing to do until I manually scrolled up.

These are huge bugs.

**Comment by Anonymadly** [comment]
I believe that you have to input a time as well in the `--:--`. When I had the issue and did that, it worked for me. I usually just put midnight, but it seems to be required.

**Comment by LoonyBeLuna** [comment]
> I believe that you have to input a time as well in the `--:--`. When I had the issue and did that, it worked for me. I usually just put midnight, but it seems to be required.

I will try it next time.

**Comment by guest2** [comment]
Can confirm--putting in a time as well as a date makes it work as expected
